corehq/apps/data_dictionary/views.py

```diff
@@ -108,6 +102,20 @@ def data_dictionary_json(request, domain, case_type_name=None):
     return JsonResponse({'case_types': props})
 
 
+def _load_fhir_resource_mappings(domain):
+    fhir_resource_types = FHIRResourceType.objects.prefetch_related('case_type').filter(domain=domain)
```

Comment:
You can prefetch all the FHIRResourceProperties at the same time ...

```suggestion
    fhir_resource_types = (
        FHIRResourceType.objects
        .select_related('case_type')
        .prefetch_related('properties__case_property')
        .filter(domain=domain)
    )
```

... and then `fhir_resource_prop_by_case_prop` would be ...

```python
    fhir_resource_prop_by_case_prop = {
        fr.case_property: fr.jsonpath
        for ft in fhir_resource_types
        for fr in ft.properties
        if fr.case_property  # NULL if not managed using Data Dictionary
    }
```

Reply:
That is a good point.

Will this still be two queries? https://stackoverflow.com/a/31237071
Thinking if the current implementation is same performance wise and also more readable?

Reply:
from reading, thought it was better to use select_related here since its 1:1.
https://github.com/dimagi/commcare-hq/pull/29336/commits/841fc5c18b214200d894b8d6282c448e8216643b